components/org.wso2.carbon.siddhi.extensions.installer.core/src/main/java/org/wso2/carbon/siddhi/extensions/installer/core/execution/DependencyRetrieverImpl.java

```diff
@@ -0,0 +1,128 @@
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Reads information related to the installation of extension dependencies. Used for retrieving extension statuses.
+ */
+public class DependencyRetrieverImpl implements DependencyRetriever {
+
+    private final Map<String, ExtensionConfig> extensionConfigs;
+
+    public DependencyRetrieverImpl(Map<String, ExtensionConfig> extensionConfigs) {
+        this.extensionConfigs = extensionConfigs;
+    }
+
+    @Override
+    public Map<String, Map<String, Object>> getAllExtensionStatuses() throws ExtensionsInstallerException {
+        Map<String, Map<String, Object>> extensionStatuses = new HashMap<>();
+        for (Map.Entry<String, ExtensionConfig> extension : extensionConfigs.entrySet()) {
+            extensionStatuses.put(extension.getKey(), getExtensionStatus(extension.getValue()));
+        }
+        return extensionStatuses;
+    }
+
+    @Override
+    public Map<String, Object> getExtensionStatusFor(String extensionId) throws ExtensionsInstallerException {
+        ExtensionConfig extension = ExtensionsInstallerUtils.findExtension(extensionId, extensionConfigs);
+        return getExtensionStatus(extension);
+    }
+
+    private Map<String, Object> getExtensionStatus(ExtensionConfig extension) throws ExtensionsInstallerException {
+        return ResponseEntityCreator.createExtensionStatusResponse(
+            extension, getExtensionInstallationStatus(extension), extension.getManuallyInstallableDependencies());
+    }
+
+    private ExtensionInstallationStatus getExtensionInstallationStatus(ExtensionConfig extension)
+        throws ExtensionsInstallerException {
+        List<DependencyConfig> dependencies = extension.getDependencies();
+        if (dependencies != null) {
+            int installedDependenciesCount = 0;
+            for (DependencyConfig dependency : dependencies) {
+                if (isDependencyInstalled(dependency)) {
+                    installedDependenciesCount++;
+                }
+            }
+            return ExtensionsInstallerUtils
+                .getInstallationStatus(installedDependenciesCount, dependencies.size());
+        }
+        throw new ExtensionsInstallerException("No dependencies were specified.");
+    }
+
+    private boolean isDependencyInstalled(DependencyConfig dependency) throws ExtensionsInstallerException {
+        String lookupRegex = dependency.getLookupRegex();
+        if (lookupRegex != null) {
+            for (UsageConfig usage : dependency.getUsages()) {
+                String usageDirectoryPath = ExtensionsInstallerUtils.getBundleLocation(usage);
+                if (!doesUsageFileExist(lookupRegex, usageDirectoryPath)) {
+                    return false;
+                }
+            }
+            return true;
+        } else {
+            throw new ExtensionsInstallerException("Unable to find property: 'lookupRegex'.");
```

Comment:
Did you mean to log the value of the variable `lookupRegex` here?

Reply:
No. The `lookupRegex` property must be given in the config of a dependency. When the property is not specified, we say `Unable to find property: 'lookupRegex'.`